psycopg3/psycopg3/connection.py

```diff
@@ -498,8 +498,28 @@ def close(self) -> None:
 
     @overload
     def cursor(
-        self, *, binary: bool = False, row_factory: Optional[RowFactory] = None
-    ) -> Cursor:
+        self,
+        *,
+        binary: bool = False,
+    ) -> Cursor[RowConn]:
+        ...
+
+    @overload
+    def cursor(
+        self,
+        *,
+        binary: bool = False,
+        row_factory: RowFactory[Row],
+    ) -> Cursor[Row]:
+        ...
+
+    @overload
+    def cursor(
+        self,
+        name: str,
+        *,
+        binary: bool = False,
```

Comment:
Black is being overzelous here. If you drop the comma off the last arguments and ask to reformat it will get these line in a better shape, like:

```python
    @overload
    def cursor(self, *, binary: bool = False) -> Cursor[RowConn]:
        ...

    @overload
    def cursor(
        self, *, binary: bool = False, row_factory: RowFactory[Row]
    ) -> Cursor[Row]:
        ...

    @overload
    def cursor(
        self, name: str, *, binary: bool = False
    ) -> ServerCursor[RowConn]:
        ...

    @overload
    def cursor(
        self, name: str, *, binary: bool = False, row_factory: RowFactory[Row]
    ) -> ServerCursor[Row]:
        ...
```